pkg/resources/builds/build.go

```diff
@@ -0,0 +1,265 @@
+	wildflyv1alpha1 "github.com/wildfly/wildfly-operator/pkg/apis/wildfly/v1alpha1"
+	"github.com/wildfly/wildfly-operator/pkg/resources"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+var log = logf.Log.WithName("wildlfyserver_buildconfigs")
+
+// GetOrCreateNewBuilderBuildConfig either returns the BuildConfig or create it
+func GetOrCreateNewBuilderBuildConfig(w *wildflyv1alpha1.WildFlyServer, client client.Client, scheme *runtime.Scheme, labels map[string]string) (*buildv1.BuildConfig, error) {
+	return getOrCreateNewBuildConfig(w, client, scheme, labels, newBuilderBuildConfig, w.Name)
+}
+
+// GetOrCreateNewRuntimeBuildConfig either returns the BuildConfig or create it
+func GetOrCreateNewRuntimeBuildConfig(w *wildflyv1alpha1.WildFlyServer, client client.Client, scheme *runtime.Scheme, labels map[string]string) (*buildv1.BuildConfig, error) {
+	return getOrCreateNewBuildConfig(w, client, scheme, labels, newRuntimeBuildConfig, w.Name+"-runtime")
+}
+
+func getOrCreateNewBuildConfig(w *wildflyv1alpha1.WildFlyServer, client client.Client, scheme *runtime.Scheme, labels map[string]string, createBuildConfig func(w *wildflyv1alpha1.WildFlyServer, labels map[string]string, name string) *buildv1.BuildConfig, name string) (*buildv1.BuildConfig, error) {
+	buildConfig := &buildv1.BuildConfig{}
+	err := resources.Get(w, types.NamespacedName{Name: name, Namespace: w.Namespace}, client, buildConfig)
+	if err != nil && !errors.IsNotFound(err) {
+		return nil, err
+	}
+	// create the buildConfig if it is not found
+	if errors.IsNotFound(err) {
+		buildConfig := createBuildConfig(w, labels, name)
+		log.Info("Create BuildConfig", "Spec", buildConfig.Spec)
+
+		if err := resources.Create(w, client, scheme, buildConfig); err != nil {
+			if errors.IsAlreadyExists(err) {
+				return nil, nil
+			}
+			return nil, err
+		}
+		return buildConfig, nil
+	}
+	// buildConfig is found, update it if it does not match the wildlfyServer generation
+	if !resources.IsCurrentGeneration(w, buildConfig) {
+		newBuildConfig := createBuildConfig(w, labels, name)
+		buildConfig.Labels = labels
+		buildConfig.Spec = newBuildConfig.Spec
+
+		if err := resources.Update(w, client, buildConfig); err != nil {
+			if errors.IsInvalid(err) {
+				// Can not update, so we delete to recreate the buildCOnfig from scratch
+				if err := resources.Delete(w, client, buildConfig); err != nil {
+					return nil, err
+				}
+				return nil, nil
+			}
+			return nil, err
+		}
+		return nil, nil
+	}
+
+	return buildConfig, nil
+
+}
+
+func newBuilderBuildConfig(w *wildflyv1alpha1.WildFlyServer, labels map[string]string, name string) *buildv1.BuildConfig {
+
+	incremental := true
+
+	builderImage := w.Spec.ApplicationSource.Source2Image.BuilderImage
```

Comment:
I think it could be useful if the Operator is able to check if the image does not exist yet and print a log in case. Someone could forget to import the image streams and do not understand why, if the CRD was deployed, the application is not build by the operator. The documentation states the images must be available though, so this is just a suggestion